### PR TITLE
#17864 fixed with test

### PR DESCRIPTION
**Pull request by wezell**
Adds a new method that populates the content map with the values of any constant fields of its content type.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java [hunk 1]
```diff
@@ -21,10 +24,14 @@
...
 import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
 import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.util.UUIDGenerator;
+import com.twelvemonkeys.util.LinkedSet;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Set'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4843681)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java [hunk 2]
```diff
@@ -166,6 +175,68 @@ public String getUrl(Contentlet contentlet) {
...
     }
+    
+    /**
+     * this test creates a widget type that has constant fields and a content of this type.
+     * It updates the constant field values and insures that the new constant field values are both
+     * in the content object and the newly transformed map
+     * @throws Exception
+     */
+    @Test
+    public void Test_Constant_fields_are_in_map() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4843681)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java [hunk 3]
```diff
@@ -166,6 +175,68 @@ public String getUrl(Contentlet contentlet) {
...
+        assertNull(map.get(constantVar1));
+        assertNull(map.get(constantVar2));
+        
+        
+        // update our Content Type constants to have values
+        final List<Field> allFields = new ArrayList<>(widgetLikeContenType.fields());
+        allFields.removeIf(f->constants.contains(f));
+        for(final Field field : constants) {
+            final String value = UUIDGenerator.generateUuid();
+            Field newField = FieldBuilder.builder(field).values(value).build();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java]: Codacy Issue found: [Local variable 'newField' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4843681)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java [hunk 4]
```diff
@@ -21,10 +24,14 @@
 import com.dotmarketing.portlets.contentlet.struts.ContentletForm;
 import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
 import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.util.UUIDGenerator;
+import com.twelvemonkeys.util.LinkedSet;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.twelvemonkeys.util.LinkedSet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4843681)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java [hunk 5]
```diff
@@ -166,6 +175,68 @@ public String getUrl(Contentlet contentlet) {
...
+        
+        // update our Content Type constants to have values
+        final List<Field> allFields = new ArrayList<>(widgetLikeContenType.fields());
+        allFields.removeIf(f->constants.contains(f));
+        for(final Field field : constants) {
+            final String value = UUIDGenerator.generateUuid();
+            Field newField = FieldBuilder.builder(field).values(value).build();
+            allFields.add(newField);
+        }
+        ContentType updatedType =APILocator.getContentTypeAPI(APILocator.systemUser()).save(widgetLikeContenType, allFields);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java]: Codacy Issue found: [Local variable 'updatedType' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4843681)

**Review by nollymar — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — COMMENTED** [review]
Very good

**Pull request merged**